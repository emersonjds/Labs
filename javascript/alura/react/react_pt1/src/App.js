import React from 'react';
import logo from './logo.svg';
import './App.css';

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <p>
          Edit Text
        </p>
        <a
          className="App-link"
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Change Project
        </a>
      </header>
    </div>
  );
}

export default App;
