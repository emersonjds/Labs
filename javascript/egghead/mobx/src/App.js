import React from "react";
import Counter from "./components/Counter";

function App() {
  return (
    <div className="App">
      <Counter></Counter>
    </div>
  );
}

export default App;
