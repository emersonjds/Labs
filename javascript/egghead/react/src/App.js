import React from "react";
import "./styles.css";

const PropTypes = {
  string(props, propName, componentName) {
    if (typeof props[propName] !== "string") {
      return new Error(
        `Hey, you should pas a string for ${propName} but you pass a ${props[propName]} for a ${componentName}`
      );
    }
  }
};

export default function App(props) {
  return (
    <div className="App">
      Hello {props.firstName} {props.lastName} !
    </div>
  );
}

App.propTypes = {
  firstName: PropTypes.string,
  lastName: PropTypes.string
};
