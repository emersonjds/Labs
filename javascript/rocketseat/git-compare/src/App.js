import React, { Fragment } from "react";
import Main from "../src/pages/Main";
import GlobalStyle from "./styles/global";

// const Title = styled.h1`
//   color: #f00;
//   font-size: "32px";
// `;

const App = () => {
  return (
    <Fragment>
      <GlobalStyle />
      <div className="App" />
      <Main />
    </Fragment>
  );
};

export default App;
