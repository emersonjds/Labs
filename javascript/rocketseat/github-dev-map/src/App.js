import React, { Component } from "react";
import MapGL, { Marker } from "react-map-gl";
import "mapbox-gl/dist/mapbox-gl.css";

export default class Map extends Component {
  state = {
    viewport: {
      width: window.innerWidth,
      height: window.innerHeight,
      latitude: -23.5439948,
      longitude: -46.6065452,
      zoom: 14
    }
  };

  componentDidMount() {
    window.addEventListener("resize", this._resize);
  }

  componentWillMount() {
    window.removeEventListener("resize", this._resize);
  }

  _resize = () => {
    this.setState({
      ...this.state.viewport({
        width: window.innerWidth,
        height: window.innerHeight
      })
    });
  };

  handleMapClick(e) {
    const [latitude, longitude] = e.lngLat;
    console.log(latitude, longitude);
  }

  render() {
    return (
      <MapGL
        {...this.state.viewport}
        onClick={this.handleMapClick}
        mapStyle="mapbox://styles/mapbox/basic-v9"
        mapboxApiAccessToken={
          "pk.eyJ1IjoiZW1lcnNvbmpkcyIsImEiOiJjanF5NGZjZHcwMDJ5M3lsandvaGlvcXBrIn0.mHlWfI4BpweXRc5sF6q1Jw"
        }
        onViewportChange={viewport => this.setState({ viewport })}
      >
        <Marker
          latitude={-23.5439948}
          longitude={-46.6065452}
          onClick={this.handleMapClick}
          captureClick={true}
        >
          <img
            style={{
              borderRadius: 100,
              width: 48,
              height: 48
            }}
            src="https://avatars2.githubusercontent.com/u/12503997?v=4"
          />
        </Marker>
      </MapGL>
    );
  }
}
