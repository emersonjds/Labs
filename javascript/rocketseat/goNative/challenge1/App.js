import './src/config/ReactotronConfig';

import React from 'react';
import { StyleSheet, Text, View } from 'react-native';

const white = '#F5FCFF';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: white,
  },
});

const App = () => (
  <View style={styles.container}>
    <Text style={styles.welcome}>Welcome to React Native!</Text>
  </View>
);

export default App;
