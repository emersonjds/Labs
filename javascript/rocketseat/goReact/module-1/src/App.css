* {
  padding: 0;
  margin: 0;
  outline: 0;
  box-sizing: border-box;
}

body {
  background-color: #f5f5f5;
  font-size: 14px sans-serif;
  color: #333;
}