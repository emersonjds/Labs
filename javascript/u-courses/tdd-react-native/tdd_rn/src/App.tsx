import React from 'react';
import Home from './screens/Home';

const App: React.FC = () => {
  return <Home />;
};

export default App;
