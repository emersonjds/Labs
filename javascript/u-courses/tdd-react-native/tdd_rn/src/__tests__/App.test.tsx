import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App';

describe('Test App', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

export {};
