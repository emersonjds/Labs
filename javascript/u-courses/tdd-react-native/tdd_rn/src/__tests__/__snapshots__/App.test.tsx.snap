// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App should render correctly 1`] = `<View />`;
